components/ws-manager/pkg/manager/monitor.go

```diff
@@ -492,9 +492,16 @@ func actOnPodEvent(ctx context.Context, m actingManager, manager *Manager, statu
 			}
 		}
 
-		_, startedDisposal := wso.Pod.Annotations[startedDisposalAnnotation]
-		span.LogKV("startedDisposal", startedDisposal)
-		if terminated && !startedDisposal {
+		var ds *workspaceDisposalStatus
+		if rawDisposalStatus, ok := pod.Annotations[disposalStatusAnnotation]; ok {
+			err := json.Unmarshal([]byte(rawDisposalStatus), ds)
+			if err != nil {
+				log.WithError(err).Errorf("cannnot parse disposalStatusAnnotation: %s", rawDisposalStatus)
+			}
+		}
+
+		span.LogKV("disposalStatusAnnotation", ds)
+		if terminated && (ds == nil || !ds.Status.IsWorking()) {
```

Comment:
This statement looks a bit hard to read now due to checking if ds is nil.
Would it make sense to instead above add something like this:
```
if ds == nil {
  ds = &workspaceDisposalStatus{}
}
```
then this if can become much cleaner:
`if terminated && !ds.Status.IsWorking() {`